rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server_queue_goals.py

```diff
@@ -73,14 +94,24 @@ def execute_callback(self, goal_handle):
 
             # Sleep for demonstration purposes
             time.sleep(1)
+        else:
```

Comment:
It's just preference, but I dislike for-else blocks, because I find them hard to parse at first glance. I would also expect inexperienced Python users to be confused too.

What if instead we wrapped the logic for starting the next goal into a function and call that just before returning the result?

Reply:
See https://github.com/ros2/examples/pull/272/commits/e4ba2ee9848ebe15ad7e75363ca8fe4629e447fc. I migrated to a try-finally block, to ensure functionality doesn't break if `execute_callback` were to `raise` (in the most general case).

Reply:
Works for me, thanks.